### PR TITLE
feat(authorization): add launch controller with redirect back to authorization uri

### PR DESCRIPTION
**Pull request by sjoulbak**
According to 5.1.1.1 and 5.1.1.2 of https://www.imsglobal.org/spec/security/v1p1#step-1-third-party-initiated-login we have to redirect the initiated third party login back. This implementation focuses on the POST request, not GET.

Note that we've added `lti_message_hint` to the authorization_uri as an optional value. This has the same purpose as the `login_hint`. As IMS sometimes make their documentation vague, this information is not found in the security documentation, but at: https://www.imsglobal.org/spec/lti/v1p3#additional-login-parameters

The idea behind having the `Lti::V1P3ToolConcern` is so single platform configurations can override the `lti_platform` and `lti_id_token_authentication` methods in order to return hardcoded values.

### CODE REVIEW

**Review by Robinoudev — CHANGES_REQUESTED** [review]
Nicely done so far 🔥

File: app/controllers/concerns/lti/v1_p3_tool_concern.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      @client ||= OpenIDConnect::Client.new(
+        identifier: lti_platform.client_id,
+        redirect_uri: lti_launch_url,
+        authorization_endpoint: lti_platform.authentication_endpoint,
+        token_endpoint: lti_platform.token_endpoint
+      )
+    end
+
+    def lti_platform
+      @lti_platform ||= Lti::Platform.find_by!(issuer: params.require(:iss), client_id: params.require(:client_id))
```
[the remark this answers is not in the document]
Comment (sjoulbak) [reply app/controllers/concerns/lti/v1_p3_tool_concern.rb:18]: Unfortunately not, this is how the attribute naming is for openid connect. For more info, please check: https://www.imsglobal.org/spec/security/v1p1#step-1-third-party-initiated-login

File: app/controllers/lti/application_controller.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# frozen_string_literal: true
+
+module Lti
+  class ApplicationController < ActionController::Base
+  end
+end
```
[the remark this answers is not in the document]
Comment (sjoulbak) [reply app/controllers/lti/application_controller.rb:6]: It is not per-se necessary for now, I'll remove it 😇

**Review by Robinoudev — APPROVED** [review]
Thanks for the detailed explanations :)

**Pull request merged**